src/main/java/com/example/spring/repository/shoppingcart/ShoppingCartRepository.java

```diff
@@ -0,0 +1,14 @@
+package com.example.spring.repository.shoppingcart;
+
+import com.example.spring.model.ShoppingCart;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface ShoppingCartRepository extends JpaRepository<ShoppingCart, Long> {
+    @Query("SELECT sc FROM ShoppingCart sc "
+            + "LEFT JOIN FETCH sc.cartItems "
+            + "WHERE sc.user.email = :email")

```

Comment:
this query is redundant here. Please use jpa method naming convention to get the method with the intended logic

Reply:
use entity graph in case you need some fields to be fetched